js/data/bucket.js

```diff
@@ -109,6 +109,35 @@ Bucket.prototype.populateArrays = function() {
 };
 
 /**
+ * Update paint arrays with the given feature properties, leaving geometries
+ * as-is.
+ * @private
+ */
+Bucket.prototype.updatePaintVertexArrays = function(programName, propertiesList) {
+    this.recalculateStyleLayers();
+
+    for (var i = 0; i < propertiesList.length; i++) {
+        var range = this._featureIndexToArrayRange[i];
+        if (!range) continue;
+        var numVertices = range.endVertex - range.startVertex + (range.endGroup - range.startGroup) * ArrayGroup.MAX_VERTEX_ARRAY_LENGTH;
+        this.prepareArrayGroup(programName, numVertices);
```

Comment:
Ah, I think I see what you mean: this wasn't an issue before, because `prepareArrayGroup` only ever needed to make enough room for one feature, whereas here, it would need to make enough room for all the features in the bucket.

👍 I'll shoot for the revision you proposed here.